optimum/exporters/onnx/__main__.py

```diff
@@ -74,20 +75,17 @@ def main():
         task, args.model, framework=args.framework, cache_dir=args.cache_dir, trust_remote_code=args.trust_remote_code
     )
 
-    if task + "-with-past" in TasksManager.get_supported_tasks_for_model_type(
+    if task != "stable-diffusion" and task + "-with-past" in TasksManager.get_supported_tasks_for_model_type(
         model.config.model_type.replace("_", "-"), "onnx"
     ):
         if args.task == "auto":  # Make -with-past the default if --task was not explicitely specified
             task = task + "-with-past"
         else:
             logger.info(
-                f"The task `{args.task}` was manually specified, and past key values will not be reused in the decoding."
-                f"Please pass `--task {args.task}-with-past` to export using the past key values."
+                f"The task `{task}` was manually specified, and past key values will not be reused in the decoding."
+                f"Please pass `--task {task}-with-past` to export using the past key values."
             )
 
-    if args.task == "auto":
-        logger.info(f"Automatic task detection to {task}.")
-
```

Comment:
I think this should be left as is, since it takes into account the `task = task + "-with-past"`

Reply:
I see, will put it back !